tests/components/habitica/test_init.py

```diff
@@ -116,6 +117,25 @@ async def test_config_entry_not_ready(
     assert config_entry.state is ConfigEntryState.SETUP_RETRY
 
 
+async def test_config_entry_auth_failed(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test config entry auth failed setup error."""
+
+    aioclient_mock.get(
+        f"{DEFAULT_URL}/api/v3/user",
+        status=HTTPStatus.UNAUTHORIZED,
+    )
+
+    config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert config_entry.state is ConfigEntryState.SETUP_ERROR
```

Comment:
I'd check that you're actually doing reauth now
```suggestion
    assert config_entry.state is ConfigEntryState.SETUP_ERROR
    
    flows = hass.config_entries.flow.async_progress()
    assert len(flows) == 1

    flow = flows[0]
    assert flow.get("step_id") == "reauth_confirm"
    assert flow.get("handler") == DOMAIN

    assert "context" in flow
    assert flow["context"].get("source") == SOURCE_REAUTH
    assert flow["context"].get("entry_id") == config_entry.entry_id
```